src/main/java/com/zufar/onlinestore/user/api/UserService.java

```diff
@@ -19,11 +22,16 @@ public class UserService implements UserApi {
 
     private final UserRepository userCrudRepository;
     private final UserDtoConverter userDtoConverter;
+    private final AuthorityService authorityService;
+    private final UserDataValidator userDataValidator;
```

Comment:
Delete this validator from this class
You should create A Custom Validator Annotation with Spring like that:
https://reflectoring.io/bean-validation-with-spring-boot/

Reply:
Done